fix: Speed Up History Cleanup ITs

yohanfernando opened the pull request:
## Description

At the moment, the ILM/ISM policies can take any time up to 5-10 minutes to apply deletion to an index. This is for cleanup tests, and we set a very low poll interval so that they run often, as a result, speeding up the tests. 


## Related issues

closes #39870

houssain-barouni commented on qa/util/src/main/java/io/camunda/qa/util/multidb/ElasticOpenSearchSetupHelper.java:
```diff
@@ -173,10 +151,51 @@ public void cleanup(final String prefix) {
+  private Map<String, Object> getSettingsWithLifecyclePollInterval(final Duration pollInterval) {
+    return Map.of(
+        "persistent",
+        Map.of("indices.lifecycle.poll_interval", String.format("%ds", pollInterval.toSeconds())));
```
since this is an Elasticsearch setting, can we make `ElasticOpenSearchSetupHelper` abstract and create an implementation for Elasticsearch (as you did for Opensearch), where we put the Elasticsearch only settings?

houssain-barouni reviewed (commented):
Great finding and good idea to reduce the test execution time! 🚀 
I am wondering though if we should run such tests that require changing cluster settings, separately in an another module or in another CI 🤔 ?

houssain-barouni reviewed (approved):
LGTM! 🚀 
I ran the AWS CI on your branch and it is passing https://github.com/camunda/camunda/actions/runs/19422657003

yohanfernando committed:
fix: Speed Up History Cleanup ITs

At the moment the ILM/ISM policies can take any time up to 5-10mins
to apply deletion to an index. This, for cleanup tests, set a very low
poll interval so that they run ofter, as a result speeding up the tests.

This refactor the current ElasticOpenSearch combined helper and create
DB specific helpers. The specific configuration is then moved to each
subclass.
